src/components/ui/Input/Input.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * This file is based on the Input component from shadcn and customized for our needs.
+ *
+ * See the official docs for more info:
+ * shadcn/ui: https://ui.shadcn.com/docs/components/input
+ */
+
+import { cn } from '@/lib/tailwind/utils';
+
+import { cva, VariantProps } from 'class-variance-authority';
+import { forwardRef, InputHTMLAttributes } from 'react';
+
+const inputVariants = cva('flex w-full text-base focus:outline-none disabled:cursor-not-allowed', {
+  variants: {
+    variant: {
+      // naming of each variant should be aligned with the design team
+      rounded: 'rounded-full bg-white border border-gray px-6 py-4 placeholder:text-gray',
+      muted: 'bg-transparent border-none placeholder:text-gray ',
+      square: 'bg-gray-lightest border-b border-primary p-2',
+    },
+    h: {
```

Comment:
I suppose naming it `height` would be better than `h` here. Here is my understanding:

When naming something, especially a variable, we should be aware of where it will be used. If its scope is small, a shorter name is usually preferred. If its scope is broad, on the other hand, the name should be easy to understand to guess what it is.

This is an example of the small scope. `f` is used only inside the `forEach`, so it's still understandable.
```ts
const foods = ['onion', 'tomato', 'meat'];
foods.forEach((f) => {
  console.log(f);
})
```

However, this `Input` component would be used outside this file, meaning it has a broader scope. We cannot tell what `h` is until we visit the file it's defined.
```
<Input variant="rounded" h="sm" />
```

Reply:
Where is the height `sm` used, by the way? I couldn't find it in Figma. We would like to avoid defining styles in code that are not defined in Figma.

Reply:
I got your point.
Just in case you missed it, my concern on this was https://github.com/genesis-tech-tribe/nishiki-frontend/pull/104#issuecomment-1859338086, but as my understanding you prefer overriding it with our original "height" property, right?

Height "sm" is now used in this square input. 
<img width="414" alt="Screenshot 2023-12-17 at 16 52 17" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99339182/57c7a8d1-6aa0-4ec8-9fb2-3dc82995bc29">

And I suppose in desktop design, there will be larger heights, so we'll add them in "height" property.

Reply:
Ohhh no, I completely missed your comment.

Reply:
> Height "sm" is now used in this square input.

Then, I suppose it's good to have two variants! Thank you!

Reply:
Ummm I'm actually changing my mind... Sorry 😅 
Overriding default "InputHTMLAttributes" is maybe inappropriate but not a big problem...
Because we're not gonna use it to be honest.

@uskayyyyy @kotaaaa 
To be honest, do you prefer using simple "height" property? It's indeed much more easy to understand.

Reply:
I do not prefer overriding to avoid unexpected behaviors.

Reply:
It actually have "size" property also

Reply:
I tried overriding but it didn't work

Reply:
Not sure but we're gonna use input type="image" maybe? To set the profile image of user, or container or something.
So maybe better to keep default height property, and use "h" for setting height?

Reply:
Thank your for always considering and testing precisely!!

I think I'll go for `h` for now!

Reply:
When we implement `type="image"`, we would create a new `<ImageImput>` component.
But as you say, I prefer not to override the default attribute.